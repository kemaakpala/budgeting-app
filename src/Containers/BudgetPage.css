.blockWrapper {
  display: flex;
  padding: 1.25rem;
}

.blockWrapper.flex-direction-column {
  flex-direction: column;
}

.alignLeft {
  margin-left: auto;
}
