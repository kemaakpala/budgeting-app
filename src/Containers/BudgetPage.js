import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlus, faTimes } from '@fortawesome/free-solid-svg-icons';
import GroupBlock from '../Components/GroupBlock';
import TransactionBlock from '../Components/TransactionBlock';
import Typography from '../Components/Typography';
import Button from '../Components/Button/Button';
import MonthUtil from '../utils/month';
import { fetchBudget } from '../store/actions'
import './BudgetPage.css';

const BudgetPage = (props) => {
  useEffect(() => {
    const { onFetchBudget } = props;
    onFetchBudget();
  }, [])

  const isLoading = props.budget.loading;
  const { data } = props.budget;

  const monthRoute = props.match.params.month;

  let returnValue = (<div>...Loading</div>)
  if (!isLoading) {
    data.forEach(dataItem => {
      if (dataItem[monthRoute]) {
        returnValue = (
          <>
            <div className="blockWrapper">
              <Typography component="h1">{MonthUtil(monthRoute)} Budget</Typography>
              <Button variant="roundedCorners" className="alignLeft" title="add transaction"><FontAwesomeIcon icon={faPlus} /></Button>
            </div>
            <div className="blockWrapper flex-direction-column">
              <GroupBlock {...dataItem[monthRoute]} />
            </div>
            <div className="blockWrapper flex-direction-column">
              {/* <div className="modal-back-drop"> */}
                {/* <div className="modal"> */}
                  {/* <Button variant="roundedCorners" className="alignLeft" ><FontAwesomeIcon icon={faTimes} /></Button> */}
                  <TransactionBlock />
                {/* </div> */}
              {/* </div> */}
            </div>
          </>
        );
      }
    })

  }

  return returnValue

}

const mapDispatchToProps = dispatch => {
  return {
    onFetchBudget: () => dispatch(fetchBudget())
  }
}
const mapStateToProps = state => {
  return {
    ...state
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(BudgetPage);