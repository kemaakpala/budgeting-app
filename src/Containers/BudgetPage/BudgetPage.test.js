import React from 'react';
import { render } from '@testing-library/react';
import BudgetPage from './BudgetPage';
import MockStoreProvider from '../../utils/mockStoreProvider';

const getMockProps = overrides => ({
  budget:{
    error: null,
    loading: true
  },
  ...overrides,
});

const renderComponent = (props = {}) => (
  <MockStoreProvider>
    <BudgetPage {...props}/>
  </MockStoreProvider>
);

describe('<BudgetPage />', () => {
  it('<BudgetPage />', () => { 
    const props = { 
      match: {
        params : {
          month: [{
            jan: '',
            feb: '',
            mar: ''
          }]
        }
      }
    };
    const { container : { firstChild } } = render( renderComponent(getMockProps({...props})) );
  
    expect(firstChild).toMatchSnapshot();
  })
})