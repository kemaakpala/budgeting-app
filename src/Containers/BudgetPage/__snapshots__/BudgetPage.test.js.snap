// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BudgetPage /> <BudgetPage /> 1`] = `
<div>
  ...Loading
</div>
`;
